include/interpolation.hxx

```diff
@@ -62,138 +62,140 @@ const T interp_to(const T& var, CELL_LOC loc, REGION region = RGN_ALL) {
   AUTO_TRACE();
   static_assert(std::is_base_of<Field2D, T>::value || std::is_base_of<Field3D, T>::value,
                 "interp_to must be templated with one of Field2D or Field3D.");
+  ASSERT1(loc != CELL_DEFAULT); // doesn't make sense to interplote to CELL_DEFAULT
 
   Mesh* fieldmesh = var.getMesh();
-  T result(fieldmesh);
 
-  if ((loc != CELL_CENTRE && loc != CELL_DEFAULT) && (fieldmesh->StaggerGrids == false)) {
+  if ((loc != CELL_CENTRE) && (fieldmesh->StaggerGrids == false)) {
     throw BoutException("Asked to interpolate, but StaggerGrids is disabled!");
   }
 
-  if (fieldmesh->StaggerGrids && (var.getLocation() != loc)) {
+  if (var.getLocation() == loc) {
+    // Nothing to do - just return unchanged
+    // Copying into result to return as returning var may increase the number of
+    // references to the var data whilst returning result doesn't
```

Comment:
This was a suggestion from @d7919 in https://github.com/boutproject/BOUT-dev/pull/921#discussion_r172671112:
> I think we may still want to copy var into result. As it stands returning var may increase the number of references to the var data whilst returning result doesn't. This may or may not be an important difference.

Reply:
(I may have been very wrong)

Reply:
I think if we did `result.ensureUnique` before returning it, that would be the case, but also a pessimisation as we would have to copy the actual data. We should be able to safely `return var`.